jgiven-spring/src/main/java/com/tngtech/jgiven/integration/spring/SpringScenarioTest.java

```diff
@@ -19,10 +16,16 @@
  *
  * @since 0.8.0
  */
-public class SpringScenarioTest<GIVEN, WHEN, THEN> extends
-        ScenarioTest<GIVEN, WHEN, THEN> implements BeanFactoryAware {
+public class SpringScenarioTest<GIVEN, WHEN, THEN> extends InternalSpringScenarioTest<GIVEN, WHEN, THEN> {
```

Comment:
I think we need to introduce a new class here, something like `RuleBasedSpringScenarioTest`. Otherwise users of old Spring versions will not be able to use that class anymore.